src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -250,7 +250,7 @@ private static unsafe void CopyImplUnBoxEachElement(Array sourceArray, int sourc
                 }
                 else if (obj is null || RuntimeHelpers.GetMethodTable(obj) != pDestMT)
                 {
-                    ThrowHelper.ThrowInvalidCastException_DownCastArrayElement();
+                    throw new InvalidCastException(SR.InvalidCast_DownCastArrayElement);
```

Comment:
I usually see throws turned into throw helpers, can you share a bit of light for my understanding of why here you're doing the opposite? Genuinely asking because I'd like to know more

Reply:
Throw helpers are performance optimization that helps in specific situations at the cost of making the code bigger or more complicated. They are typically worth it for perf critical methods where they help with making the methods inlineable or with making the method frame smaller, and for generic methods where they help with making the per-instantiation code size smaller. None of these cases apply here. As with any performance optimization, you can always measure when in doubts.

Reply:
Thank you for the explanation!